common/endpoints/endpoints.go

```diff
@@ -15,7 +15,12 @@ func BaseURLIam(region string) string {
 	if region == "" {
 		log.Fatal(errors.New("empty region supplied, can't generate IAM URL"))
 	}
-	return fmt.Sprintf("https://iam.%s.otc.t-systems.com:443/v3", region)
+	switch region {
+	case "eu-ch2":
+		return fmt.Sprintf("https://iam-pub.eu-ch2.sc.otc.t-systems.com:443/v3")
```

Comment:
I think swapping fmt.Sprintf out for fmt.Sprint should be enough to solve this :D